webapp/src/js/bootstrapper/index.js

```diff
@@ -32,46 +33,54 @@
   };
 
   const getDbInfo = function() {
-    // parse the URL to determine the remote and local database names
-    const location = window.location;
     const dbName = 'medic';
-    const port = location.port ? ':' + location.port : '';
-    const remoteDB = location.protocol + '//' + location.hostname + port + '/' + dbName;
     return {
       name: dbName,
-      remote: remoteDB
+      remote: `${utils.getBaseUrl()}/${dbName}`
     };
   };
 
   var getLocalDbName = function(dbInfo, username) {
     return dbInfo.name + '-user-' + username;
   };
 
+  const setReplicationId = (POUCHDB_OPTIONS, localDb) => {
+    return localDb.id().then(id => {
+      POUCHDB_OPTIONS.remote_headers['medic-replication-id'] = id;
+    });
+  };
+
   var initialReplication = function(localDb, remoteDb) {
     setUiStatus('LOAD_APP');
     var dbSyncStartTime = Date.now();
     var dbSyncStartData = getDataUsage();
-    var replicator = localDb.replicate
-      .from(remoteDb, {
-        live: false,
-        retry: false,
-        heartbeat: 10000,
-        timeout: 1000 * 60 * 10, // try for ten minutes then give up
-      });
 
-    replicator
-      .on('change', function(info) {
-        console.log('initialReplication()', 'change', info);
-        setUiStatus('FETCH_INFO', { count: info.docs_read || '?' });
-      });
+    return serverSidePurge
+      .info()
+      .then(info => {
+        const replicator = localDb.replicate
+          .from(remoteDb, {
+            live: false,
+            retry: false,
+            heartbeat: 10000,
+            timeout: 1000 * 60 * 10, // try for ten minutes then give up,
+            query_params: { initial_replication: true }
+          });
 
-    return replicator
-      .then(function() {
-        var duration = Date.now() - dbSyncStartTime;
+        replicator
+          .on('change', function(info) {
+            console.log('initialReplication()', 'change', info);
```

Comment:
Accidentally left behind?

Reply:
Argh, you made me doubt myself! No, not left behind, part of `master` code. 
https://github.com/medic/medic/blame/4c70aca3a3024cd6e233ffb3b74c72d8bfb0b02b/webapp/src/js/bootstrapper/index.js#L64